src/sentry/sentry_metrics/querying/stats.py

```diff
@@ -0,0 +1,137 @@
+
+
+def _get_mql_string(aggregate: str, group_by: Sequence[str]) -> str:
+
+    # TODO(metrics): add support for reason tag
+    group_by_tags = []
+    if "outcome" in group_by:
+        group_by_tags.append("outcome.id")
+    if "project" in group_by:
+        group_by_tags.append("project_id")
+
+    if group_by_tags:
+        return f"{aggregate} by ({', '.join(group_by_tags)})"
+
+    return aggregate
+
+
+def _run_metrics_outcomes_query(
+    start, end, group_by, interval, organization, projects, environments
+) -> dict[str, list]:
+
+    mql_string = _get_mql_string(METRIC_OUTCOME_AGGREGATE, group_by)
+
+    rows = run_queries(
+        mql_queries=[MQLQuery(mql_string)],
+        start=start,
+        end=end,
+        interval=int(3600 if interval is None else interval.total_seconds()),
+        organization=organization,
+        projects=projects,
+        environments=environments,
+        referrer=Referrer.OUTCOMES_TIMESERIES.value,
+    ).apply_transformer(MetricsOutcomesTransformer())
+
+    return rows
+
+
+def _run_estimated_metrics_hours_query(
+    start, end, group_by, organization, projects, environments
+) -> dict[str, list]:
+    mql_string = _get_mql_string(METRIC_HOURS_AGGREGATE, group_by)
+
+    rows = run_queries(
+        mql_queries=[MQLQuery(mql_string)],
+        start=start,
+        end=end,
+        # metrics hours queries have to be bucketed by 1h
+        interval=int(3600),
+        organization=organization,
+        projects=projects,
+        environments=environments,
+        referrer=Referrer.OUTCOMES_TIMESERIES.value,
+    ).apply_transformer(MetricsOutcomesTransformer())
+
+    return rows
+
+
+def _run_metrics_cardinality_query(
+    start, end, group_by, interval, organization, projects, environments
+):
+
+    if not interval:
+        interval = 3600
+    elif interval.total_seconds() < 3600:
+        interval = 3600
+    else:
+        interval = interval.total_seconds()
+
+    mql_string = _get_mql_string(METRIC_CARDINALITY_AGGREGATE, group_by)
+
+    rows = run_queries(
+        mql_queries=[MQLQuery(mql_string)],
+        start=start,
+        end=end,
+        interval=interval,
+        organization=organization,
+        projects=projects,
+        environments=environments,
+        referrer=Referrer.OUTCOMES_TIMESERIES.value,
+    ).apply_transformer(MetricsOutcomesTransformer())
+
+    return rows
+
+
+def run_metric_stats_query(
+    query: QueryDict,
+    organization: Organization,
+    projects: Sequence[Project],
+    environments: Sequence[Environment],
+):
+
+    start, end = get_date_range_from_params(query)
+    group_by = query.getlist("groupBy", [])
+    interval = parse_stats_period(query.get("interval", "1h"))
+
+    category = query.get("category")
+    # TODO(metrics): remove metrics and metricsSeconds after FE is updated
+    if category == "metrics" or category == "metricSecond" or category == "metricsOutcomes":
```

Comment:
metricsOutcomes is not called in the endpoint - is this on purpose?

Reply:
good catch! thx!